mapentity/static/mapentity/mapentity.filter.js

```diff
@@ -87,6 +68,60 @@ MapEntity.TogglableFilter = L.Class.extend({
         return $(this.popover.data('bs.popover').tip);
     },
 
+    load_filter_form: function (mapsync) {
+        var self = this;
+        // On first click on Filter button, load HTML content for form
+        if (!self.loaded_form) {
+            var filter_url = $('#mainfilter').attr('filter-url');
+            $.get(filter_url)
+                .done(response => {
+                    $('#filters-panel .filter-spinner-container').hide();
+                    // Replace simple form that contains only BBOX with full form, including attributes
+                    $('#mainfilter').replaceWith(response);
+                    // Update L.MapListSync to refresh datatable on change
+                    mapsync.options.filter.form = $('#mainfilter');
+                    // Bind new form buttons to keep refreshing list on click
+                    $("#filter").click(function(e) {
+                        mapsync._onFormSubmit(e);
+                    });
+                    $("#reset").click(function(e) {
+                        mapsync._onFormReset(e);
+                    });
+                    // Bind setfields to update list of enabled fields displayed on hover
+                    $('#mainfilter').find('select,input').change(function () {
+                        self.setfield(this);
+                    });
+                    self.loaded_form = true;
+
+                    // Use chosen for multiple values
+                    $("form#mainfilter").bind("reset", function() {
+                        setTimeout(function() {
+                            $('form#mainfilter select[multiple]').trigger('chosen:updated');
+                        }, 1);
+                    });
+
+                    // Make sure filter-set class is added if a choice is selected.
+                    $('#mainfilter select[multiple]').chosen().on('change', function (e) {
+                        var $target = $(e.target),
+                            name = $target.attr('name'),
+                            $container = $('div#id_' + name + '_chzn > ul');
+                        if ($(e.target).find('option:selected').length > 0) {
+                            $container.addClass('filter-set');
+                        }
+                        else {
+                            $container.removeClass('filter-set');
+                        }
```

Comment:
You can use `toggleClass` with a second parameter as bool to add/remove the className.

```suggestion
                        $container.toggleClass('filter-set', $target.find('option:selected').length > 0);
```